xarray/backends/rasterio_.py

```diff
@@ -131,15 +133,18 @@ def open_rasterio(filename, chunks=None, cache=None, lock=None):
         raise ValueError('Unknown dims')
     coords['band'] = np.asarray(riods.indexes)
 
-    # Get geo coords
+    # Get coordinates
+    if LooseVersion(rasterio.__version__) < '1.0':
+        transform = riods.affine
+    else:
+        transform = riods.transform
```

Comment:
Can we use attribute lookup fallback instead, e.g.,
```
try:
    transform = riods.transform
except AttributeError:
    transform = riods.affine
```

That's usually slightly more robust than explicitly checking versions.

Reply:
The `transform` attribute exists on the older version as well, but there it is a tuple specifying the transform in GDAL format. Maybe it'd be better to check whether `transform` is an instance of `Affine` instead?

Reply:
> Maybe it'd be better to check whether transform is an instance of Affine instead?

I don't know... Reading the transform attribute with rio < 1.0 raises a warning, reading the affine attribute with rio >= 1.0 raises a warning, so we'll have to catch the warnings in order to test this. Would that be more elegant?